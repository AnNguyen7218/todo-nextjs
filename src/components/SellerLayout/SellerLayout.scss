@import 'config.media';
@import 'config.variables';

.layout-content {
  min-height: calc(100vh - #{$header-height-sm} - #{$footer-height-sm});

  @include break(md) {
    min-height: calc(100vh - #{$header-height-md} - #{$footer-height-md});
  }

  @include break(lg) {
    min-height: calc(100vh - #{$header-height-lg} - #{$footer-height-lg});
  }

  @include break(xl) {
    min-height: calc(100vh - #{$header-height-xl} - #{$footer-height-xl});
  }
}
