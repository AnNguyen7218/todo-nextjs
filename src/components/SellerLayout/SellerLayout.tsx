import React from 'react';
import ToastProvider from 'contexts/ToastContext/ToastProvider';
import { Footer } from 'components';
import css from './SellerLayout.scss';

type Props = {
  children: React.ReactNode;
};

const SellerLayout: React.FC<Props> = ({ children }) => {
  return (
    <ToastProvider>
      <>
        <div className={css['layout-content']}>{children}</div>
        <Footer />
      </>
    </ToastProvider>
  );
};

export default SellerLayout;
