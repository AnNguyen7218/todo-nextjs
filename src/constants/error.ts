export const ERROR_MESSAGE = {
  SERVER_ERROR: 'Server error, please try again.',
  REQUIRED: '*Required field',
  INVALID_EMAIL: 'Invalid email address',
  TAKEN_EMAIL: 'Email already exists',
  PASSWORD_NO_MATCH: "Password doesn't match",
  INVALID_PASSWORD:
    'Must contain at least 8 characters with upper and lower case and numbers',
  USER_NOT_EXIST:
    "Sorry, we couldn't find an account with that email. Please enter another email or sign up for a new account.",
};

const AUTH_FORM_ERROR_FIRSTNAME_PROVIDE = 'Auth.form.error.firstName.provide';
const AUTH_FORM_ERROR_LASTNAME_PROVIDE = 'Auth.form.error.lastName.provide';
const AUTH_FORM_ERROR_EMAIL_PROVIDE = 'Auth.form.error.email.provide';
const AUTH_FORM_ERROR_EMAIL_TAKEN = 'Auth.form.error.email.taken';
const AUTH_FORM_ERROR_EMAIL_FORMAT = 'Auth.form.error.email.format';
const AUTH_FORM_ERROR_PASSWORD_PROVIDE = 'Auth.form.error.password.provide';
const AUTH_FORM_ERROR_PASSWORD_FORMAT = 'Auth.form.error.password.format';
const AUTH_FORM_ERROR_USER_NOT_EXIST = 'Auth.form.error.user.not-exist';
const AUTH_FORM_ERROR_PASSWORD_CONFIRMATION_PROVIDE =
  'Auth.form.error.passwordConfirmation.provide';
const AUTH_FORM_ERROR_CODE_PROVIDE = 'Auth.form.error.code.provide';
const AUTH_FORM_ERROR_PASSWORD_MATCHING = 'Auth.form.error.password.matching';
const AUTH_FORM_ERROR_PARAMS_PROVIDE = 'Auth.form.error.params.provide';

export type ServerErrorID =
  | typeof AUTH_FORM_ERROR_FIRSTNAME_PROVIDE
  | typeof AUTH_FORM_ERROR_LASTNAME_PROVIDE
  | typeof AUTH_FORM_ERROR_EMAIL_PROVIDE
  | typeof AUTH_FORM_ERROR_EMAIL_TAKEN
  | typeof AUTH_FORM_ERROR_EMAIL_FORMAT
  | typeof AUTH_FORM_ERROR_PASSWORD_PROVIDE
  | typeof AUTH_FORM_ERROR_PASSWORD_CONFIRMATION_PROVIDE
  | typeof AUTH_FORM_ERROR_CODE_PROVIDE
  | typeof AUTH_FORM_ERROR_PASSWORD_MATCHING
  | typeof AUTH_FORM_ERROR_PARAMS_PROVIDE
  | typeof AUTH_FORM_ERROR_PASSWORD_FORMAT;

export const SERVER_ERROR_MAP = {
  [AUTH_FORM_ERROR_FIRSTNAME_PROVIDE]: {
    field: 'firstName',
    message: ERROR_MESSAGE.REQUIRED,
  },
  [AUTH_FORM_ERROR_LASTNAME_PROVIDE]: {
    field: 'lastName',
    message: ERROR_MESSAGE.REQUIRED,
  },
  [AUTH_FORM_ERROR_EMAIL_PROVIDE]: {
    field: 'email',
    message: ERROR_MESSAGE.REQUIRED,
  },
  [AUTH_FORM_ERROR_EMAIL_TAKEN]: {
    field: 'email',
    message: ERROR_MESSAGE.TAKEN_EMAIL,
  },
  [AUTH_FORM_ERROR_EMAIL_FORMAT]: {
    field: 'email',
    message: ERROR_MESSAGE.INVALID_EMAIL,
  },
  [AUTH_FORM_ERROR_PASSWORD_PROVIDE]: {
    field: 'password',
    message: ERROR_MESSAGE.REQUIRED,
  },
  [AUTH_FORM_ERROR_PASSWORD_FORMAT]: {
    field: 'password',
    message: ERROR_MESSAGE.INVALID_PASSWORD,
  },
  [AUTH_FORM_ERROR_USER_NOT_EXIST]: {
    field: 'form', // will show on top of form
    message: ERROR_MESSAGE.USER_NOT_EXIST,
  },
  [AUTH_FORM_ERROR_PASSWORD_CONFIRMATION_PROVIDE]: {
    field: 'passwordConfirmation',
    message: ERROR_MESSAGE.REQUIRED,
  },
  [AUTH_FORM_ERROR_PASSWORD_MATCHING]: {
    field: 'passwordConfirmation',
    message: ERROR_MESSAGE.PASSWORD_NO_MATCH,
  },
  [AUTH_FORM_ERROR_CODE_PROVIDE]: {
    field: 'form',
    message: '',
  },
  [AUTH_FORM_ERROR_PARAMS_PROVIDE]: {
    field: 'form',
    message: '',
  },
};
