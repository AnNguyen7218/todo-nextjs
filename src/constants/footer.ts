import FB from 'assets/icons/ic-facebook.svg';
import Insta from 'assets/icons/ic-insta.svg';
import Twitter from 'assets/icons/ic-twitter.svg';
import LinkIn from 'assets/icons/ic-linkin.svg';

export const FOOTER_LINKS_ICON: FooterLinkIcon[] = [
  {
    url: '/#',
    icon: FB,
  },
  {
    url: '/#',
    icon: Insta,
  },
  {
    url: '/#',
    icon: Twitter,
  },
  {
    url: '/#',
    icon: LinkIn,
  },
];

export const FOOTER_LINKS_TEXT: FooterLinkText[] = [
  {
    url: '/#',
    title: 'Term',
  },
  {
    url: '/#',
    title: 'Privacy',
  },
];
