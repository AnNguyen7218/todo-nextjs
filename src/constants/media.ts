export const MEDIA_MD_UP = '(min-width: 720px)';
export const MEDIA_LG_UP = '(min-width: 960px)';

export const LG = 960;
export const XL = 1200;
