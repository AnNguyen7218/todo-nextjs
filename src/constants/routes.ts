export const ROUTES = {
  HOME: '/',
  SIGN_UP: '/sign-up',
  SIGN_IN: '/sign-in',
  EMAIL_CONFIRMATION: '/email-confirmation',
  FORGOT_PASSWORD: '/forgot-password',
  CHANGE_PASSWORD: '/change-password',
  GETTING_STARTED: '/getting-started',
  EMAIL_CONFIRMATION_SUCCESS: '/email-confirmation-success',
  EMAIL_CONFIRMATION_FAILURE: '/email-confirmation-failure',
  PROFILE: '/profile',
};
