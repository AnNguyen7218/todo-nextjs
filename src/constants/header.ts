// import { logout } from 'utils/auth';
import { ROUTES } from 'constants/routes';

export const MENU_ITEMS: MenuItemType[] = [
  {
    title: 'item1',
  },
  {
    title: 'item2',
  },
  {
    title: 'item3',
  },
  {
    title: 'item4',
  },
  {
    title: 'item5',
  },
];

export const USER_MENU: UserMenuType[] = [
  {
    title: 'Your profile',
    type: 'link',
    href: ROUTES.PROFILE,
  },
  {
    title: 'Log out',
    type: 'action',
    // action: logout,
  },
];
