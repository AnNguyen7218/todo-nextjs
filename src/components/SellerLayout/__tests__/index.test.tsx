import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { ROUTES } from 'constants/routes';
import SellerLayout from '../SellerLayout';

afterEach(cleanup);

describe('SellerLayout Component', () => {
  test('SellerLayout should match snapshot', () => {
    jest.spyOn(require('next/router'), 'useRouter').mockImplementation(() => ({
      pathname: ROUTES.HOME,
    }));
    const { container } = render(
      <SellerLayout>
        <p>Seller Layout</p>
      </SellerLayout>
    );
    expect(container).toMatchSnapshot();
  });

  test('SellerLayout should match snapshot when current route is signup page', () => {
    jest.spyOn(require('next/router'), 'useRouter').mockImplementation(() => ({
      pathname: ROUTES.SIGN_UP,
    }));
    const { container } = render(
      <SellerLayout>
        <p>Seller Layout</p>
      </SellerLayout>
    );
    expect(container).toMatchSnapshot();
  });

  test('SellerLayout should match snapshot when current route is login page', () => {
    jest.spyOn(require('next/router'), 'useRouter').mockImplementation(() => ({
      pathname: ROUTES.SIGN_IN,
    }));
    const { container } = render(
      <SellerLayout>
        <p>Seller Layout</p>
      </SellerLayout>
    );
    expect(container).toMatchSnapshot();
  });
});
