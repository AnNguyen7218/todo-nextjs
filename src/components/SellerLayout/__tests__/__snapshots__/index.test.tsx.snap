// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellerLayout Component SellerLayout should match snapshot 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="container container"
  >
    <ic-logo.svg
      classname="logo--dark"
      height="27"
      width="100"
    />
    <div
      class="right-container"
    >
      <div
        class="desktop-menu-group"
      >
        <div
          class="container"
        >
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item1
            </div>
          </button>
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item2
            </div>
          </button>
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item3
            </div>
          </button>
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item4
            </div>
          </button>
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item5
            </div>
          </button>
        </div>
      </div>
      <a
        class="button button--fixed button--containedColor button--black button-try"
        href="/sign-up"
      >
        <div
          class="button-children"
        >
          Join us
        </div>
      </a>
      <ic-hamburger-menu.svg
        classname="hamburger-menu"
        data-testid="toggle-button"
      />
      <div
        class="dropdown-container"
      >
        <div
          class="items-container"
        >
          <div
            class="items-section"
          >
            <div
              class="menu-item"
            >
              item1
            </div>
            <div
              class="menu-item"
            >
              item2
            </div>
            <div
              class="menu-item"
            >
              item3
            </div>
            <div
              class="menu-item"
            >
              item4
            </div>
            <div
              class="menu-item"
            >
              item5
            </div>
          </div>
          <div
            class="items-section"
          >
            <a
              class="link"
              href="/sign-in"
            >
              <div
                class="menu-item"
              >
                <div
                  class="menu-item-title menu-user-title"
                >
                  Log in
                </div>
              </div>
            </a>
          </div>
          <div
            class="items-section"
          >
            <div
              class="menu-item"
            >
              Help center
            </div>
            <div
              class="menu-item"
            >
              <ic-language.svg
                classname="ic-language"
              />
              <span>
                ENG
              </span>
            </div>
          </div>
        </div>
      </div>
      <a
        class="button button--fixed button--textOnly desktop-sign-in"
        href="/sign-in"
      >
        <div
          class="button-children"
        >
          Log in
        </div>
      </a>
    </div>
  </div>
  <div
    class="layout-content"
  >
    <p>
      Seller Layout
    </p>
  </div>
  <div
    class="wrapper"
  >
    <div
      class="container container"
    >
      <div
        class="top-section"
      >
        <ic-logo.svg
          classname="logo--light"
          height="27"
          width="100"
        />
        <div
          class="icons-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-facebook.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-insta.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-twitter.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-linkin.svg
              classname="icon"
            />
          </a>
        </div>
      </div>
      <div
        class="bottom-section"
      >
        <p
          class="sub-title"
        >
          Â© 2020 Intrepid
        </p>
        <div
          class="texts-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Term
            </span>
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Privacy
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SellerLayout Component SellerLayout should match snapshot when current route is login page 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="container container container--auto-hidden"
  >
    <ic-logo.svg
      classname="logo--dark"
      height="27"
      width="100"
    />
    <div
      class="right-container"
    >
      <div
        class="desktop-menu-group"
      >
        <div
          class="container"
        >
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item1
            </div>
          </button>
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item2
            </div>
          </button>
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item3
            </div>
          </button>
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item4
            </div>
          </button>
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item5
            </div>
          </button>
        </div>
      </div>
      <a
        class="button button--fixed button--containedColor button--black button-try"
        href="/sign-up"
      >
        <div
          class="button-children"
        >
          Join us
        </div>
      </a>
      <ic-hamburger-menu.svg
        classname="hamburger-menu"
        data-testid="toggle-button"
      />
      <div
        class="dropdown-container"
      >
        <div
          class="items-container"
        >
          <div
            class="items-section"
          >
            <div
              class="menu-item"
            >
              item1
            </div>
            <div
              class="menu-item"
            >
              item2
            </div>
            <div
              class="menu-item"
            >
              item3
            </div>
            <div
              class="menu-item"
            >
              item4
            </div>
            <div
              class="menu-item"
            >
              item5
            </div>
          </div>
          <div
            class="items-section"
          >
            <a
              class="link"
              href="/sign-in"
            >
              <div
                class="menu-item"
              >
                <div
                  class="menu-item-title menu-user-title"
                >
                  Log in
                </div>
              </div>
            </a>
          </div>
          <div
            class="items-section"
          >
            <div
              class="menu-item"
            >
              Help center
            </div>
            <div
              class="menu-item"
            >
              <ic-language.svg
                classname="ic-language"
              />
              <span>
                ENG
              </span>
            </div>
          </div>
        </div>
      </div>
      <a
        class="button button--fixed button--textOnly desktop-sign-in"
        href="/sign-in"
      >
        <div
          class="button-children"
        >
          Log in
        </div>
      </a>
    </div>
  </div>
  <div
    class="container container"
  >
    <button
      class="button button--fixed button--textOnly back-button"
      type="button"
    >
      <div
        class="button-children back-button-content"
      >
        <ic-back.svg
          classname="icon-back"
        />
      </div>
    </button>
    <a
      class="button button--fixed button--textOnly right-button"
      href="/sign-up"
    >
      <div
        class="button-children right-btn-children"
      >
        Sign up
      </div>
    </a>
  </div>
  <div
    class="layout-content"
  >
    <p>
      Seller Layout
    </p>
  </div>
  <div
    class="wrapper auto-hidden"
  >
    <div
      class="container container"
    >
      <div
        class="top-section"
      >
        <ic-logo.svg
          classname="logo--light"
          height="27"
          width="100"
        />
        <div
          class="icons-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-facebook.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-insta.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-twitter.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-linkin.svg
              classname="icon"
            />
          </a>
        </div>
      </div>
      <div
        class="bottom-section"
      >
        <p
          class="sub-title"
        >
          Â© 2020 Intrepid
        </p>
        <div
          class="texts-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Term
            </span>
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Privacy
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SellerLayout Component SellerLayout should match snapshot when current route is signup page 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="container container container--auto-hidden"
  >
    <ic-logo.svg
      classname="logo--dark"
      height="27"
      width="100"
    />
    <div
      class="right-container"
    >
      <div
        class="desktop-menu-group"
      >
        <div
          class="container"
        >
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item1
            </div>
          </button>
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item2
            </div>
          </button>
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item3
            </div>
          </button>
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item4
            </div>
          </button>
          <button
            class="button button--fixed button--textOnly menu-item-container"
            type="button"
          >
            <div
              class="button-children"
            >
              item5
            </div>
          </button>
        </div>
      </div>
      <a
        class="button button--fixed button--containedColor button--black button-try"
        href="/sign-up"
      >
        <div
          class="button-children"
        >
          Join us
        </div>
      </a>
      <ic-hamburger-menu.svg
        classname="hamburger-menu"
        data-testid="toggle-button"
      />
      <div
        class="dropdown-container"
      >
        <div
          class="items-container"
        >
          <div
            class="items-section"
          >
            <div
              class="menu-item"
            >
              item1
            </div>
            <div
              class="menu-item"
            >
              item2
            </div>
            <div
              class="menu-item"
            >
              item3
            </div>
            <div
              class="menu-item"
            >
              item4
            </div>
            <div
              class="menu-item"
            >
              item5
            </div>
          </div>
          <div
            class="items-section"
          >
            <a
              class="link"
              href="/sign-in"
            >
              <div
                class="menu-item"
              >
                <div
                  class="menu-item-title menu-user-title"
                >
                  Log in
                </div>
              </div>
            </a>
          </div>
          <div
            class="items-section"
          >
            <div
              class="menu-item"
            >
              Help center
            </div>
            <div
              class="menu-item"
            >
              <ic-language.svg
                classname="ic-language"
              />
              <span>
                ENG
              </span>
            </div>
          </div>
        </div>
      </div>
      <a
        class="button button--fixed button--textOnly desktop-sign-in"
        href="/sign-in"
      >
        <div
          class="button-children"
        >
          Log in
        </div>
      </a>
    </div>
  </div>
  <div
    class="container container"
  >
    <button
      class="button button--fixed button--textOnly back-button"
      type="button"
    >
      <div
        class="button-children back-button-content"
      >
        <ic-back.svg
          classname="icon-back"
        />
      </div>
    </button>
    <a
      class="button button--fixed button--textOnly right-button"
      href="/sign-in"
    >
      <div
        class="button-children right-btn-children"
      >
        Login
      </div>
    </a>
  </div>
  <div
    class="layout-content"
  >
    <p>
      Seller Layout
    </p>
  </div>
  <div
    class="wrapper auto-hidden"
  >
    <div
      class="container container"
    >
      <div
        class="top-section"
      >
        <ic-logo.svg
          classname="logo--light"
          height="27"
          width="100"
        />
        <div
          class="icons-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-facebook.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-insta.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-twitter.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-linkin.svg
              classname="icon"
            />
          </a>
        </div>
      </div>
      <div
        class="bottom-section"
      >
        <p
          class="sub-title"
        >
          Â© 2020 Intrepid
        </p>
        <div
          class="texts-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Term
            </span>
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Privacy
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
